### PR TITLE
Switch maintanance

### PR DESCRIPTION
**Pull request by RSDuck**
* The old memory allocator has become nonfunctional due to breaking changes in [libnx](https://github.com/switchbrew/libnx). Since `--d:nimAllocPagesViaMalloc` works just as well, I have removed the custom allocator entirely.
* Command line arguments on Switch are contrary what the documentation previously stated supported when using the nxlink tool to sent binaries to the Switch, so I enabled support for them.

Should I document some/all of these changes in the changelog?

### CODE REVIEW

**Comment by Clyybber** [comment]
> Should I document some/all of these changes in the changelog?

Yeah, I think that's a good idea :)

**Comment by RSDuck** [comment]
should be all done

File: doc/nimc.rst [hunk 1]
```diff
@@ -406,16 +406,18 @@ to your usual `nim c`:cmd: or `nim cpp`:cmd: command and set the `passC`:option:
 and `passL`:option: command line switches to something like:
 
 .. code-block:: cmd
-  nim c ... --passC="-I$DEVKITPRO/libnx/include" ...
+  nim c ... --d:nimAllocPagesViaMalloc --gc:arc --passC="-I$DEVKITPRO/libnx/include" ...
```
Comment (Araq) [inline doc/nimc.rst]: ``--gc:orc`` works too and casual users are better off with a cycle collector.

File: doc/nimc.rst [hunk 2]
```diff
@@ -406,16 +406,18 @@ to your usual `nim c`:cmd: or `nim cpp`:cmd: command and set the `passC`:option:
...
   --passL="-specs=$DEVKITPRO/libnx/switch.specs -L$DEVKITPRO/libnx/lib -lnx"
 
 or setup a ``nim.cfg`` file like so::
 
   #nim.cfg
+  --gc:arc
```
Comment (Araq) [inline doc/nimc.rst]:
```suggestion
  --gc:orc
```

**Pull request merged**